### ISSUE TITLE
Emacs mode: return to the beginning of the line

### ISSUE THREAD

**Issue by francoisthire** [Deducteam/lambdapi#397]
Assume I have this line
```
constant symbol typ : TYPE
```
If I press return my cursor goes back to 

```
constant symbol typ : TYPE
                                   ^
```

Which is counter-intuitive. All the major modes I use, by default return to the beginning of the line in particular if this line is the last one of a file.

**Comment by fblanqui** [comment]
Yes. @vycastor will try to improve this in the next weeks.

**Comment by gabrielhdt** [comment]
Yeah, but it's really related to the SMIE we use for indentation. If we want to get rid of this behaviour, we must change the indentation engine. The SMIE cannot know that the term is finished after `TYPE`.

Well in the case of `TYPE`, we can make a special case since it's a constant. But you'll have the same problem with `constant symbol appl: T z`, if `T` is a dependent type.

We may be able to exchange information between SMIE and the lsp server, I don't know. I don't even know if it has been done anywhere. It might be interesting though.

**Comment by gabrielhdt** [comment]
Another solution is `M-x electric-indent-mode`.

**Comment by Rehan-MALAK** [comment]
> Another solution is `M-x electric-indent-mode`.

Don't know exactly why, but `M-x paragraph-indent-minor-mode` worked for me with emacs+evil
